### PR TITLE
fix: Refactor usages of unsafe

### PR DESCRIPTION
**Pull request by Troy-Butler**
Addresses issue #95

### CODE REVIEW

**Review by timraymond — DISMISSED** [review]

File: pkg/utils/utils_linux.go [hunk 1]
```diff
@@ -78,7 +77,7 @@ func HostToNetShort(i uint16) uint16 {
 func determineEndian() binary.ByteOrder {
 	var endian binary.ByteOrder
 	buf := [2]byte{}
-	*(*uint16)(unsafe.Pointer(&buf[0])) = uint16(0xABCD)
+	binary.BigEndian.PutUint16(buf[:], uint16(0xABCD))
```
[the remark this answers is not in the document]
Comment (Troy-Butler) [reply pkg/utils/utils_linux.go]: Done!

**Review by anubhabMajumdar — CHANGES_REQUESTED** [review]
Can you add `Testing Done` section in description?

**Comment by rbtr** [comment]
@anubhabMajumdar are you willing to take this now that E2E is running?

**Comment by anubhabMajumdar** [comment]
> @anubhabMajumdar are you willing to take this now that E2E is running?

@Troy-Butler Can you update the description with testing done for `advanced` mode and paste the advanced `TCP` metrics generated? Either local or remote context would work. This would ensure your code changes are tested. Thanks!

**Comment by Troy-Butler** [comment]
> > @anubhabMajumdar are you willing to take this now that E2E is running?
> 
> @Troy-Butler Can you update the description with testing done for `advanced` mode and paste the advanced `TCP` metrics generated? Either local or remote context would work. This would ensure your code changes are tested. Thanks!

Please forgive my ignorance, this is my first contribution to this project. What do I need to run to generate the metrics you need?

**Comment by anubhabMajumdar** [comment]
> > > @anubhabMajumdar are you willing to take this now that E2E is running?
> > 
> > 
> > @Troy-Butler Can you update the description with testing done for `advanced` mode and paste the advanced `TCP` metrics generated? Either local or remote context would work. This would ensure your code changes are tested. Thanks!
> 
> Please forgive my ignorance, this is my first contribution to this project. What do I need to run to generate the metrics you need?

Apologies, I should have included the instructions. Here's how you can get started:
- Build container image with your changes - https://retina.sh/docs/contributing/#build
- Deploy in advance mode - https://retina.sh/docs/installation/setup#advanced-mode-with-local-context-with-capture-support
- Check the `*adv_*` metrics - https://retina.sh/docs/contributing/#metrics

Another way to quickly test a single plugin on your host machine (in your case that would be Packetparser) would be this - https://retina.sh/docs/contributing/#build (look for `To build binary of a plugin and test it`).

Here's the corresponding main file you need to run - https://github.com/microsoft/retina/blob/main/test/plugin/packetparser/main_linux.go .
You would also need to build and run - https://github.com/microsoft/retina/tree/main/init/retina .

Let me know if you need any help, would be happy to help! Thanks again for your contribution.

**Comment by Troy-Butler** [comment]
Is there anything else I need to do?

**Pull request closed without merging**